### PR TITLE
APIS-7647 transform empty object into map[string]string

### PR DESCRIPTION
**Pull request by miladmikeal**

### CODE REVIEW

File: generate/struct_test.go [hunk 1]
```diff
@@ -375,6 +375,7 @@ func TestGoType(t *testing.T) {
 		array       bool
 		required    bool
 		pointers    bool
+		emptyObject bool
```
Comment (gannett-ggreer) [inline generate/struct_test.go:378]: I would add `fields` as a test field and then put in your sample string field to the existing test entry so it keeps working and let yours be `nil` and thus 0.

File: generate/struct.go [hunk 2]
```diff
@@ -418,8 +418,12 @@ func (ef *extractedField) goType(required, pointers bool) string {
 			customType = true
 		}
 	case "object":
-		// This only happens with nested structures.
-		goType = "struct"
+		if len(ef.fields) == 0 {
+			goType = "map[string]string"
```
Comment (gannett-ggreer) [inline generate/struct.go:422]: I'd add a TODO comment about using the `additionalProperties` JSON schema value to more intelligently create the target type in the future.

**Comment by gannett-ggreer** [comment]
When it hits this line:

https://github.com/GannettDigital/jstransform/blob/master/generate/struct.go#L66

you're probably going to get output like:
```
    field map[string]string {
```

**Comment by adamhostettler** [comment]
I think it would be good to see a real-world example using this branch, as a proof of concept. Doing so can also help provide insight into potential bugs.

**Comment by miladmikeal** [comment]
> I think it would be good to see a real-world example using this branch, as a proof of concept. Doing so can also help provide insight into potential bugs.

I plan on testing these changes directly onto my `content-api-shared-datatypes` changes using the replace directive before actually merging.

**Pull request closed without merging**